src/Transports/MassTransit.AmazonSqsTransport/AmazonSqsTransport/AmazonSqsReceiveContext.cs

```diff
@@ -91,6 +92,7 @@ public Task ValidateLockStatus()
         public override void Dispose()
         {
             _activeTokenSource.Dispose();
```

Comment:
why just not to `Cancel` it here? so the loop will exit cleanly?

Reply:
in general `isDisposed` indicates that some operations order is incorrect and masks real issue

Reply:
> why just not to Cancel it here? so the loop will exit cleanly?

That was my first idea, too. However see:
https://stackoverflow.com/a/12474762 and https://stackoverflow.com/a/30855490

Given that .NET Framework is still supported by MassTransit, it seems to be the safer option to cleanly dispose the token source.

> in general isDisposed indicates that some operations order is incorrect and masks real issue

True, but as `RenewMessageVisibility` runs in an own task in the background, you don't really have control over that (when the Task runs and when not).

An alternative would be to just wait for the Task to be completed when completing the receive context, eg:
```cs
public async Task Complete()
{
    _activeTokenSource.Cancel();
    await _renewMessageVisibiltyTask;

    return _clientContext.DeleteMessage(_settings.EntityName, _message.ReceiptHandle);
}

public async Task Faulted(Exception exception)
{
    _activeTokenSource.Cancel();
    await _renewMessageVisibiltyTask;

    // .... the rest of the implementation 
}
```

Reply:
>> why just not to Cancel it here? so the loop will exit cleanly?
>
> That was my first idea, too. However see https://stackoverflow.com/a/30855490
>
> Given that .NET Framework is still supported by MassTransit, it seems to be the safer option to cleanly dispose the token source.

Sorry, got confused with the versioning here a bit.

> If, however, you're creating your CTS with CreateLinkedTokenSource, or if you're using the CTS' timer functionality, it can be more impactful to use Dispose.

Both isn't the case, so just cancelling the CTS instead of disposing it should be fine. Your call.

Reply:
we need to dispose it anyway, we need to `await` a task, we have similar here: https://github.com/MassTransit/MassTransit/blob/master/src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaMessageConsumer.cs#L163-L172

Reply:
Fair enough, just a heads up - I have an issue in the backlog that I just bumped to the next release to rewrite this method to be a little cleaner (based upon how Azure renews locks in their SDK). So this will likely be replaced with the new method.

Reply:
there is might be a better option, use `IAsyncDisposable` for amazon context, which is gonna call base.
but then you need to use `DisposeAsync` in Consumer

Reply:
So with `IAsyncDisposable ` you can `Cancel` token, then await task and do other job